src/lua/compat.lua

```diff
@@ -69,12 +68,7 @@ local options = {
         obsolete = nil,
         brief = JSON_ESCAPE_BRIEF,
         run_action_now = true,
-        action = function(is_new)
-            local esc_slash = not is_new
-            require('json').cfg{encode_escape_forward_slash = esc_slash}
-            internal.json_escape_forward_slash_toggle(esc_slash)
-            internal.msgpuck_escape_forward_slash_toggle(esc_slash)
```

Comment:
How about code that uses `mp_snprint()` (and so `mp_char2escape` from `src/lib/msgpuck/hints.c`)?

Reply:
AFAICS `mp_snprint` is an internal function used for debugging and logging only. It dumps MsgPack data in a human-readable format, which isn't JSON. I don't see why we should enable escaping of the forward slash for it. AFAIU it was enabled solely because of `json_encode` (which is used by the JSON logger) using the `mp_char2escape` table internally. This doesn't make any sense, anyway - `json_encode` should use the same escape table as our JSON encoder, not the one from the MsgPack library. I tried to explain that in the last commit message.

Reply:
`log.info()` is also used for debugging and logging only, however it is deliberately changed by the compat option. I'm worrying about the inconsistency. When the new behavior will be enabled in compat, some messages will have escaped forward slash after this patchset (while some will conform to the option).

Reply:
`log.info` doesn't use `mp_snprint`. It uses the `json.encode` Lua function if its argument is a table. It also uses the `json_encode` C function internally in `say.c` if the log format is JSON. Both functions now don't have anything to do with `mp_char2escape` and whether they escape forward slashes or not is configured with the new tweak.

BTW the effect that the compat option has on `log.info` is tested:

https://github.com/tarantool/tarantool/blob/d4e928091b18406cd966e54f9eb207d3682ae730/test/app-luatest/gh_6200_lua_cjson_escapes_slash_test.lua#L72-L127

Reply:
Discussed with @alyapunov and @Totktonada and agreed that it's okay to remove forward slash escaping from `mp_snprint` unconditionally. We don't need to do it under the compat option, because we only use it for presenting MsgPack data in *unspecified* human-readable format (e.g. for logging) - we never promised that it won't change (we didn't even promise that it's a valid JSON).

I opened a PR for libmsgpuck: https://github.com/tarantool/msgpuck/pull/25